### ISSUE TITLE
SIM Toolkit displayed in Chinese - no way to change it

### ISSUE THREAD

**Issue by AL2** [CyanogenDefy/android_device_motorola_jordan#53]
Like the tittle says, the SIM Toolkit is only displayed in Chinese on my phone [and I can't use it since I can't tell what the options are].

My Defy was bought in China (ME525) but it was set to English from day one AND I did flash Nordic SBF before using my current setup: CM7-RC1v2.

So there is a problem with the application to correctly detect local language setting.

Cheers.

**Comment by AL2** [comment]
After flashing Nordic sbf again today, I've noticed that the SIM tool is also in Chinese on that 2.2 ROM even though language is set to English-US.

So problem with Nordic sbf?

**Comment by maniac103** [comment]
Your SIM is a chinese one, though, right? If I'm not completely mistaken, all texts displayed in the Stk app are provided by the SIM, not the OS.

**Comment by AL2** [comment]
Thanks Maniac,

I believe that you're right [at least on the Chinese SIM card part ;-)] so if the apk is really just throwing info from the SIM, then I guess it's a non issue then.

ty again.

**Comment by maniac103** [comment]
Well, seemingly there is a way for the SIM card to request the user's language from the OS (the PROVIDE_LOCAL_INFORMATION command). I've added support for this by porting a commit from CAF: https://github.com/CyanogenDefy/android_frameworks_base/commit/b44a1164bc4cba1ba9637060e0558018d7d8ab04

This fix should be in the next nightlies. Please report back whether it helped (as your SIM needs to utilize it). If it doesn't help, I really think there isn't much we can do.

**Comment by jaskle** [comment]
I think only the Chinese SIMtool have expensive items...
I delete it every time
I heard these comments will be sent to the owner of the mailbox, I hope no bother.

**Comment by AL2** [comment]
I've tgried on 110713-0601, 110718-0731 and 110719-0102 nightly builds and no luck!  ;-(

But that begs the question: would it be possible to had a switch in "Contacts" (Import/Export contacts) to add the option to save contact(s) to SIM card?

Because like I wrote on xda, for any new contact I have, I can only only chose to create contact under Google account or "Phone-only, unsynched". Which the later only saves it onto the phone; not the SIM card....

I could (but don't want to) synch with Google or add the contact to phone and then export all contacts to SD card. 
But that ain't very convenient when I need to use my SIM card into another phone...

Thanks for trying anyway Mister Baumanniac.
;-)

Edit:

Or, are you aware of any application that can be installed to manage contacts and SIM?

Thanks again.

/ A L
